File: ShopTask/Frontend/app/order/models/product.model.ts
export class Product {
    private _id: number;
    private _title: string;
    private _price: number;
    private _description: string;
    private _category: string;

    public get id(): number {
        return this._id;
    }

    public get title(): string {
        return this._title;
    }

    public get price(): number {
        return this._price;
    }

    public get description(): string {
        return this._description;
    }

    public get category(): string {
        return this._category;
    }

    constructor(data: any){
        this._id = data.Id;
        this._title = data.Title;
        this._price = data.Price;
        this._description = data.Description;
        this._category = data.Category;
    }
};

File: ShopTask.Application/Models/ProductModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web.Mvc;
using ShopTask.Resources.Products;

namespace ShopTask.Application.Models
{
    public class ProductModel
    {
        public int? Id { get; set; }

        [Required]
        public string Title { get; set; }

        [Display]
        [RegularExpression(@"((^[1-9][0-9]*(\.[0-9]*)?)|(^0\.(([0-9]*)?[1-9]([0-9]*)?)))$")]
        public decimal Price { get; set; }

        public string Description { get; set; }

        public int CategoryId { get; set; }

        public CategoryModel Category { get; set; }
    }
}